### PR TITLE
EWL-3815 js conflicts

### PR DESCRIPTION
**Pull request by froboy**
## Ticket(s)

Please do not submit a Pull Request without a relevant ticket! If you are creating a new pattern or feature, create an issue describing the need for this feature Github **first** and then link to it below.

**Github Issue**

NA

**Jira Ticket**

- [EWL-3815: Topics | SG | Resolve jQuery duplication](https://issues.ama-assn.org/browse/EWL-3815)

## Description

This work entails two pieces:
1. Creating a new `drupal.js` file which segments the custom JS from jQuery libraries to prevent conflicts in Drupal.
1. Wrapping the existing JS in drupal behaviors and adding a library to the style guide to accomodate those wrappers. This work follows directly from https://github.com/palantirnet/butler/pull/41

## To Test

- `gulp serve`
- Compare any component that uses JS with that on https://americanmedicalassociation.github.io/AMA-style-guide
- Observe nothing breaks at all definitely.
- Read the new section in README.md.

## Relevant Screenshots/GIFs

NA

## Remaining Tasks

This has a corresponding Drupal ticket https://github.com/AmericanMedicalAssociation/ama-d8/pull/17

## Additional Notes

Anything more to add?

### CODE REVIEW

**Comment by froboy** [comment]
I merged the drupal-develop changes in and pushed to the new branch. I got a git error the first time that seemed like some gunk in my `/.public` directory. `rm -rf .public` then `gulp drupal-deploy` succeeded.

**Comment by rpama** [comment]
@froboy It seems the `drupal-deploy` command is creating a `styleguide/.publish` directory. Can we add this to our `.gitignore` or do we need to make it part of the repo? (That's not directly related to this PR, but I just now noticed it).

**Comment by froboy** [comment]
@rpama yup... i think Sourcetree was missing it because it's a hidden directory, but i see it when i run `git status`. I was also in the process of adding a step to clean the directory before publishing to stop possible git issues. The above should fix that.

**Comment by rpama** [comment]
The links to the Butler docs seem to be broken or not public. I can't get to what you linked in the Readme:
https://github.com/palantirnet/butler/blob/drupal-attach-behaviors/docs/JS.md
but I can get to
https://github.com/palantirnet/butler/blob/master/docs/JS.md

Likewise, the JS template link inside _that_ JS doc is also not working, so maybe we want to link the template ourselves separately?
Does not work:
https://github.com/palantirnet/butler/blob/drupal-attach-behaviors/docs/JS_TEMPLATE.js
Works:
https://github.com/palantirnet/butler/blob/master/docs/JS_TEMPLATE.js

File: README.md [hunk 1]
```diff
@@ -104,3 +104,6 @@ For example, to use the side-by-side Two Up Layout, add the <code>layout_two_up<
 Note: you should not need to nest elements with class <code>.layout</code>.
 
 The <code>.layout</code> and <code>.layout_number_up</code> classes can be added to container elements both in individual patterns (i.e. organisms) and in the Twig templates that include them (e.g. templates). In general, if a specific pattern can be assumed to always use the same layout, it's fine to include the layout within that pattern.
+
```
Comment (rpama) [inline README.md]: As per discussion in the last FEDs meeting, development documentation belongs in docs folders. So this should probably be added to the Javascript section of the `docs/code-conventions` file, not in the Readme.

**Comment by froboy** [comment]
@rpama I've submitted https://github.com/palantirnet/butler/pull/87 to correct the link in that documentation. Working on a PR to bring that stuff in, as it does need to be edited a little for our context.

**Comment by rpama** [comment]
This looks great, @froboy. I will leave it to you to do the merge on Monday since I'm on my way out, but you have my approval!

**Comment by rpama** [comment]
MERGED @froboy

**Pull request merged**